errors.go

```diff
@@ -0,0 +1,40 @@
+package decimal
+
+import "fmt"
+
+// ErrorFractionLimit is returned when the decimal exponent exceed int32 range.
+type ErrorFractionLimit struct {
+	value string
+}
+
+// Error implements error interface.
+func (e *ErrorFractionLimit) Error() string {
+	return fmt.Sprintf("can't convert %s to decimal: fractional part too long", e.value)
+}
+
+// ErrorInvalidFormat is returned when the input string is not valid integer.
+type ErrorInvalidFormat struct {
+	reason string
+}
+
+// Error implements error interface.
+func (e *ErrorInvalidFormat) Error() string {
+	return e.reason
+}
+
+// ErrorInvalidType is returned when the value passed into sql.Scanner is not
+// with expected type. (valid types: int64, float64, []byte, string)
+type ErrorInvalidType struct {
+	reason string
+}
+
+// Error implements error interface.
+func (e *ErrorInvalidType) Error() string {
+	return e.reason
+}
+
+func assertErrorInterface() {
+	var _ error = (*ErrorFractionLimit)(nil)
+	var _ error = (*ErrorInvalidFormat)(nil)
+	var _ error = (*ErrorInvalidType)(nil)
```

Comment:
Is there any reason to put them in a function?

Reply:
Style taken from emirpasic/gods project. I think this explains the intension of these code, although it may be quite straightforward in golang.